docs/stability.md

```diff
@@ -0,0 +1,97 @@
+
+The deprecation note SHOULD point out viable alternatives.
+
+## Interface Changes
+
+Public interfaces from `kube` is currently allowed to change between **minor** versions, provided github release & [[changelog]] provides adequate guidance on the change, and the amount of user facing changes is minimized and trivialised. In particular:
+
+- changes needed to user code should have a [diff code comment](https://github.com/kube-rs/kube-rs/releases/tag/0.73.0) on the change
+- changes as a result of interface changes should be explained
+- if changes affect controller-rs or version-rs, link to the fixing commit
+- when renaming functions, removing arguments, changing imports, show what to search/replace
+
+In general; we **prefer deprecations and duplication** of logic **if** it **avoids confusion**. But sometimes the least confusing thing to do is often just removing the old interface.
+
+## Major Release 1.0
+
+While we satisfy [client requirements](https://github.com/kubernetes/design-proposals-archive/blob/main/api-machinery/csi-new-client-library-procedure.md#client-support-level) for stability, we have not released a `1.0.0` version yet.
+
+This is due to an unstable subset of advanced functionality, and one outstanding feature:
+
+1. [protobuf serialization layer](https://github.com/kube-rs/kube-rs/issues/725) is unimplemented
+2. advanced client usage with websockets is new and still experiencing light shifting
+3. runtime features are improving and still shifting
+
+Those are the primary reasons for not releasing a 1.0, but there are also external concerns:
+
+- we depend on pre-1.0 libraries and upgrading these under semver would force major bumps
+- the public async iterator interface is [still being stabilised in rust](https://github.com/rust-lang/rust/issues/79024)
+
+As a result, we instead:
+
+- **track** our breaking changes with [changelog-change labeled PRs](https://github.com/kube-rs/kube-rs/pulls?q=is%3Apr+label%3Achangelog-change+is%3Aclosed)
+- **inform** on necessary changes in [releases](https://github.com/kube-rs/kube-rs/releases) and the [[changelog]]
+- **carefully change** according to our [guidelines on interface changes](#interface-changes)
+- **deprecate** according to our [deprecation strategy](#deprecation-strategy)
+
+## Post 1.0
+
+After 1.0, we hope to follow Kubernetes' versioning with a skew-policy identical to [client-go](https://github.com/kubernetes/client-go#compatibility-matrix), using planned setup in [k8s-pb](https://github.com/kube-rs/k8s-pb/issues/10).
+
+This means that it will be clearer what versions to pick, but it also means we are unable to follow semver's requirement for [major version increments on breaking changes](https://semver.org/spec/v1.0.0.html#spec-item-10), and technically breaking changes like upgrading pre-1.0 libraries can sneak in between versions.
```

Comment:
This isn't really negotiable IMO, Cargo's dependency resolution is built around the assumption that Semver applies. If that means that we end up releasing v100 at some point then so be it.

(Ideally we won't get that far any time soon, but obviously some degree of breakage is kind of unavoidable eventually.)

Reply:
Oh shit, that is true. That actually means we cannot follow kubernetes versioning.

Reply:
Have removed the more speculative bits about 1.0 and deferred to the issue. Will update that later.